calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseCalendarService.java

```diff
@@ -1636,6 +1639,51 @@ public String merge(String siteId, Element root, String archivePath, String from
 		return results.toString();
 	}
 
+	/*
+	 * Look at the properties of the event and determine if it should be merged or ignored
+	 */
+	public boolean shouldMergeEvent(Element el)
+	{
+		NodeList nodeList = el.getElementsByTagName("property");
+
+		for (int i = 0; i < nodeList.getLength(); i++) {
+			Element prop = (Element) nodeList.item(i);
+			String name = prop.getAttribute("name");
+			String value = prop.getAttribute("value");
+			if ("BASE64".equalsIgnoreCase(prop.getAttribute("enc")))
+			{
+				value = Xml.decodeAttribute(prop, "value");
+			}
+
+			if ( StringUtils.contains(name, CalendarConstants.TOOL_ID) &&
+				(StringUtils.contains(value, "sakai.assignment.grades") || StringUtils.contains(value, "sakai.forums") ||
```

Comment:
AssignmentConstants.TOOL_ID and DiscussionForumService.FORUMS_TOOL_ID here

Reply:
Thanks for all - I made the changes.